### ISSUE TITLE
optimizer.add_param_group添加的参数无法更新

### ISSUE THREAD

**Issue by 376498485** [PaddlePaddle/X2Paddle#613]
我做了一个简单的测试，我使用了X2Paddle里的optimzer.Adam，首先用w的参数创建优化器，然后使用optimzer.add_param_group()添加b的参数，训练前后w的参数更新了，但b的参数没有更新，这是怎么回事呢？
如下是结果和代码：
![image](https://user-images.githubusercontent.com/49092533/121028427-67034e80-c7da-11eb-93ab-4957247929ec.png)
```python
import paddle
import paddle.nn as nn
from lib.optimizer import Adam

class Weight(nn.Layer):
    def __init__(self):
        super().__init__()
        self.l1 = nn.Linear(1, 1)
    def forward(self, x):
        x = self.l1(x)
        return x

class Bias(nn.Layer):
    def __init__(self):
        super().__init__()
        self.bias = paddle.create_parameter([1], dtype='float32')
    def forward(self, x):
        return x + self.bias

w = Weight()
b = Bias()

optim = Adam(params=w.parameters(), lr=0.1)
optim.add_param_group({'params': b.parameters()})

x = paddle.to_tensor(2.0)
y = x
output = b(w(x)) # output = w * x + b
print("训练前的参数：#############")
print("w:\n", w.parameters())
print("b:\n", b.parameters())
loss = y-output
loss.backward()
optim.step()
optim.clear_grad()
print("训练后的参数：#############")
print("w:\n", w.parameters())
print("b:\n", b.parameters())

**Comment by SunAhong1993** [comment]
我们暂未碰到过这样的使用方式，经查看当前X2Padlde实现Adam有一定问题，这边修复后回复您。
